### PR TITLE
Added string.trim, string.trim{Left,Right,StartEnd}

### PR DESCRIPTION
**Pull request by Perryvw**
Also string.trimLeft, string.trimRight, string.trimStart and string.trimEnd

### CODE REVIEW

File: test/unit/builtins/string.spec.ts [hunk 1]
```diff
@@ -265,3 +265,21 @@ test.each([
         `;
     expect(util.transpileAndExecute(code)).toBe(6);
 });
+
+const trimTestCases = [
```
Comment (ark120202) [inline test/unit/builtins/string.spec.ts]: I haven't checked the spec, but from a quick look at MDN page it also has to handle `\r\n\t\f\v\uFEFF\xA0`

Comment (Perryvw) [reply test/unit/builtins/string.spec.ts]: right, I'll add some more testcases

File: src/lualib/StringTrimStart.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+function __TS__StringTrimStart(this: string): string {
+    // http://lua-users.org/wiki/StringTrim
+    const [result] = string.gsub(this, "^[%s\xA0\u{FEFF}]*", "");
```
Comment (ark120202) [inline src/lualib/StringTrimStart.ts]:
```suggestion
    const [result] = string.gsub(this, "^[%s\xA0\uFEFF]*", "");
```

File: test/unit/builtins/string.spec.ts [hunk 3]
```diff
@@ -265,3 +265,28 @@ test.each([
...
+    "",
+    " ",
+    "   ",
+    "foo ",
+    "foo ",
```
Comment (ark120202) [inline test/unit/builtins/string.spec.ts]: Are these 2 strings same?

Comment (Perryvw) [reply test/unit/builtins/string.spec.ts]: One is a tab and one is a space, but alignment makes them look the same.

File: test/unit/builtins/string.spec.ts [hunk 4]
```diff
@@ -265,3 +265,28 @@ test.each([
...
+    "    foo",
+    "   foo     ",
+    "    foo     ",
+    "  foo    bar     ",
+    "\r\nfoo\n\r\n",
+    "\r\nfoo\nbar\n\r\n",
+];
+describe.each(["trim", "trimEnd", "trimRight", "trimStart", "trimLeft"])("string.%s", trim => {
+    test.each(trimTestCases)("matches JS result (%p)", testString => {
+        util.testExpression`\`${testString}\`.${trim}()`.expectToMatchJsResult();
```
Comment (ark120202) [inline test/unit/builtins/string.spec.ts]: Would it work with

```suggestion
        util.testExpression`${util.formatCode(testString)}.${trim}()`.expectToMatchJsResult();
```

**Pull request merged**